stubs/seaborn/seaborn/palettes.pyi

```diff
@@ -0,0 +1,149 @@
+from collections.abc import Iterable, Sequence
+from typing import TypeVar, overload
+from typing_extensions import Literal, Self, TypeAlias
+
+from matplotlib.colors import Colormap, LinearSegmentedColormap, ListedColormap
+from matplotlib.typing import ColorType
+
+__all__ = [
+    "color_palette",
+    "hls_palette",
+    "husl_palette",
+    "mpl_palette",
+    "dark_palette",
+    "light_palette",
+    "diverging_palette",
+    "blend_palette",
+    "xkcd_palette",
+    "crayon_palette",
+    "cubehelix_palette",
+    "set_color_codes",
+]
+
+_T = TypeVar("_T")
+
+SEABORN_PALETTES: dict[str, list[str]]
+MPL_QUAL_PALS: dict[str, int]
+QUAL_PALETTE_SIZES: dict[str, int]
+QUAL_PALETTES: list[str]
+
+class _ColorPalette(list[_T]):
+    def __enter__(self) -> Self: ...
+    def __exit__(self, *args: object) -> None: ...
+    def as_hex(self) -> _ColorPalette[str]: ...
+
+_RGBColorPalette: TypeAlias = _ColorPalette[tuple[float, float, float]]
+_SeabornPaletteName: TypeAlias = Literal[
+    "deep", "deep6", "muted", "muted6", "pastel", "pastel6", "bright", "bright6", "dark", "dark6", "colorblind", "colorblind6"
+]
+
+@overload
+def color_palette(  # type: ignore[misc]
+    palette: _SeabornPaletteName | None = None, n_colors: int | None = None, desat: float | None = None, *, as_cmap: Literal[True]
+) -> list[str]: ...  # this might be a bug in seaborn because we expect the return type to be a Colormap instance
+@overload
+def color_palette(
+    palette: str | Sequence[ColorType], n_colors: int | None = None, desat: float | None = None, *, as_cmap: Literal[True]
+) -> Colormap: ...
+@overload
+def color_palette(
+    palette: str | Sequence[ColorType] | None = None,
+    n_colors: int | None = None,
+    desat: float | None = None,
+    as_cmap: Literal[False] = False,
+) -> _RGBColorPalette: ...
+@overload
+def hls_palette(
+    n_colors: int = 6, h: float = 0.01, l: float = 0.6, s: float = 0.65, *, as_cmap: Literal[True]
+) -> ListedColormap: ...
+@overload
+def hls_palette(
+    n_colors: int = 6, h: float = 0.01, l: float = 0.6, s: float = 0.65, as_cmap: Literal[False] = False
+) -> _RGBColorPalette: ...
+@overload
+def husl_palette(
+    n_colors: int = 6, h: float = 0.01, s: float = 0.9, l: float = 0.65, *, as_cmap: Literal[True]
+) -> ListedColormap: ...
+@overload
+def husl_palette(
+    n_colors: int = 6, h: float = 0.01, s: float = 0.9, l: float = 0.65, as_cmap: Literal[False] = False
+) -> _RGBColorPalette: ...
+@overload
+def mpl_palette(name: str, n_colors: int = 6, *, as_cmap: Literal[True]) -> LinearSegmentedColormap: ...
```

Comment:
Technically this and similar overloads on nearby functions are incomplete, as they don't cover calls like `mpl_palette("x", 1, True)`. You'd need another overload to cover that case.

Reply:
You are right. I deliberately left out the third overload here because I think in this case it is OK for the stub to be a little bit stricter than runtime and to expect a keyword argument especially that a usage like `mpl_palette("x", 1, True)` feels cryptic. But if you prefer I can add the missing overload.

Reply:
OK, let's just keep it as is as we can always change it later. Agree these really should be keyword-only.